crates/noirc_evaluator/src/ssa/conditional.rs

```diff
@@ -272,7 +272,7 @@ impl DecisionTree {
         let mut ins = ctx[left].instructions.clone();
         ins.extend(&ctx[right].instructions);
         let mut modified = false;
-        super::optim::cse_block(ctx, left, &mut ins, &mut modified);
+        super::optim::cse_block(ctx, left, &mut ins, &mut modified).unwrap();
```

Comment:
Why are we unwrapping here? An error originating from here seems reasonable if there was a constrain that was found to be always false inside.

Reply:
isn't unwrap triggering an error in that case ?

Reply:
It would be panicking with an error instead of reporting it to the user in noirc_driver/src/lib.rs line 168

Reply:
return an error now